statsmodels/multivariate/tests/test_ml_factor.py

```diff
@@ -78,12 +78,33 @@ def test_exact_em():
             s = np.sqrt(np.diag(c))
             c /= np.outer(s, s)
             fa = Factor(corr=c, n_factor=n_factor, method='ml')
-            load_e, uniq_e = fa._fit_ml_em(200)
+            load_e, uniq_e = fa._fit_ml_em(2000)
             c_e = np.dot(load_e, load_e.T)
             c_e.flat[::c_e.shape[0]+1] += uniq_e
             assert_allclose(c_e, c, rtol=1e-4, atol=1e-4)
 
 
+def test_fit_ml_em_random_state():
+    """Ensure Factor._fit_ml_em doesn't change numpy's singleton random state
+
+    This test arose from https://github.com/statsmodels/statsmodels/issues/7357
+    """
+
```

Comment:
alternative would be `initial = np.random.get_state()`
then later
```
final = np.random.get_state()
```
and check that initial was equal to `final`.
 
But your method gets the point across.

Reply:
I like that better too. Should have looked more into this. Just changed the test to do it this way.